test/Microsoft.TestPlatform.AcceptanceTests/TranslationLayerTests/DiscoverTests.cs

```diff
@@ -227,13 +227,13 @@ public void CancelTestDiscovery(RunnerInfo runnerInfo)
         // Act
         var discoveryTask = Task.Run(() => _vstestConsoleWrapper.DiscoverTests(testAssemblies, GetDefaultRunSettings(), discoveryEvents.Object));
 
-        Task.Delay(1000).Wait();
+        Task.Delay(2000).Wait();
```

Comment:
This seems flaky, does this need to cancel on some particular test? Maybe you can cancel after getting first batch of tests / first batch of tests from a given source that has many of them?

Reply:
There is no specific needs but putting the cancellation after the discovery of some sources makes the test really slow locally (around 30s) for the net451 TFM

![image](https://user-images.githubusercontent.com/11340282/158181629-cb1f00a3-76a1-433f-929d-55eef1ef181f.png)

Reply:
I have updated the test to have a better logic and prevent flakiness.